src/libcore/iter/traits/exact_size.rs

```diff
@@ -69,7 +69,8 @@
 /// ```
 #[stable(feature = "rust1", since = "1.0.0")]
 pub trait ExactSizeIterator: Iterator {
-    /// Returns the exact number of times the iterator will iterate.
+    /// Returns the exact length of the iterator, which is the number of times
+    /// the iterator will return `Some(T)` before returning `None`.
```

Comment:
This is a *little* long for a summary sentence. I'm not sure of the best way to re-word it though. It's likely that some of this will have to go below.

Reply:
What about:
```markdown
Returns the exact length of the iterator.

The implementation ensures that the iterator will return exactly `len()` times a `Some(T)` value, before returning `None`.